product_sequence/models/product_product.py

```diff
@@ -32,7 +32,8 @@ def create(self, vals):
     @api.multi
     def write(self, vals):
         for product in self:
-            if product.default_code in [False, '/']:
+            if (product.default_code in [False, '/'] and vals.get(
+                    'default_code', '/') == '/'):
```

Comment:
@gdgellatly Thanks for your review.

However I disagree with your suggestion :

>  It seems very unlikely both those conditions would be met except for the very specific behaviour the commit is for

The second condition I added is always True if 'default_code' key isn't in vals dict, and as the first condition doesn't change, I don't understand why it seems so unlikely...

>  For example if a default_code already existed and it was removed, this test evaluates False yet in this situation we would expect a pass.

This is already the case actually with this module, my PR doesn't pretend to address this behaviour.

>  Also if a default_code did not exist, and we were writing something else, say uom_id this test also fails where it would be expected to pass.

If we want to write on uom_id it will be done anyway : 
- Should the product already have a default_code, this default_code won't change, and the new uom_id will be written as is the case before the PR. 
- Should the product be without a default_code and the default_code isn't in vals dict, a new one will be written from the sequence as was previously the case with this module too. 

So I don't think it's a great idea to call super first and make another update afterwards. Moreover should you write False on default_code, it would be done through super() but then will be overwritten by the sequence, so it doesn't improve anything.